### PR TITLE
ETFE-2960 Duty paid user's access - feature

### PR DESCRIPTION
**Pull request by grahampaulcook**

### CODE REVIEW

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Ran against JT's and they fail

File: test/controllers/errors/DutyPaidUnauthorisedControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+      s"being rendered for lang '${viewMessages.lang.code}'" must {
+
+        s"return UNAUTHORIZED ($UNAUTHORIZED)" in new Test {
+
+          val result = controller.unauthorised()(fakeRequest)
+
+          status(result) mustBe UNAUTHORIZED
+
+          Html(contentAsString(result)) mustBe view()(fakeRequest, messages(fakeRequest))
```
[the remark this answers is not in the document]
Comment (grahampaulcook) [reply test/controllers/errors/DutyPaidUnauthorisedControllerSpec.scala]: https://github.com/hmrc/emcs-tfe-frontend/pull/59/commits/90a9542cc8d47a86fbc34bad7de6f7fbefcd5d3b

File: conf/messages [hunk 2]
```diff
@@ -253,6 +253,12 @@ errors.unauthorised.title = Unauthorised
 errors.unauthorised.heading = Unauthorised
 errors.unauthorised.message = You are not authorised to access this service
 
+errors.unauthorised.dutypaiduser.title = There is a problem
+errors.unauthorised.dutypaiduser.heading = There is a problem
+errors.unauthorised.dutypaiduser.message1 = Duty paid traders cannot yet use the Excise Movement and Control System (EMCS).
+errors.unauthorised.dutypaiduser.message2 = HMRC will notify you when you are able to use the service.
+errors.unauthorised.dutypaiduser.message3 = If you think you have signed in with the wrong details, sign out and check the details you have are correct.
```
[the remark this answers is not in the document]
Comment (grahampaulcook) [reply conf/messages]: Agreed thats clearer - https://github.com/hmrc/emcs-tfe-frontend/pull/59/commits/f491a2fdfa90d2218bce132815caff3399a7adb6

File: app/controllers/predicates/AuthAction.scala [hunk 3]
```diff
@@ -94,15 +95,23 @@ class AuthActionImpl @Inject()(override val authConnector: AuthConnector,
                                                 internalId: String,
                                                 credId: String
                                                )(block: UserRequest[A] => Future[Result])
-                                               (implicit request: Request[A]): Future[Result] =
+                                               (implicit request: Request[A]): Future[Result] = {
+
+    def dutyPaidUser: Boolean = ernFromUrl.startsWith("XIPA") || ernFromUrl.startsWith("XIPB")
```
[the remark this answers is not in the document]
Comment (grahampaulcook) [reply app/controllers/predicates/AuthAction.scala]: https://github.com/hmrc/emcs-tfe-frontend/pull/59/commits/dc1c5d0042b54a886c35e7e26a895c9170aa06ef

**Comment by grahampaulcook** [comment]
> Ran against JT's and they fail

JT's have now been updated under https://jira.tools.tax.service.gov.uk/browse/ETFE-2967

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
One issue I spotted was that the page title doesn't have the usual suffix:
image

**Comment by grahampaulcook** [comment]
> One issue I spotted was that the page title doesn't have the usual suffix: image

It matches the Unauthorised page title
image

**Review by LewisMitchell — COMMENTED** [review]
Just waiting on JT's to be reviewed. There was one failure when running all the test suites (JT's)

**Comment by grahampaulcook** [comment]
> Just waiting on JT's to be reviewed. There was one failure when running all the test suites (JT's)

I only ran `run_tests_acc.sh`
https://github.com/hmrc/emcs-tfe-journey-tests/pull/341 approved.

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**